dashboard/test/controllers/pd/regional_partner_controller_test.rb

```diff
@@ -0,0 +1,44 @@
+require 'test_helper'
+
+class Pd::RegionalPartnerControllerTest < ActionController::TestCase
+  self.use_transactional_test_case = true
+  setup_all do
+    @regional_partner = create :regional_partner
+    @workshop_admin = create :workshop_admin
+  end
+
+  def self.test_workshop_admin_only(method, action, params = nil)
+    test_user_gets_response_for action, user: :student, method: method, params: params, response: :forbidden
```

Comment:
You can collapse the first 4 into one call that gets repeated over an array consisting of :student, :teacher, :facilitator, :workshop_organizer